wpilibj/src/main/java/edu/wpi/first/wpilibj/Filesystem.java

```diff
@@ -0,0 +1,58 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2008-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj;
+
+import java.io.File;
+
+/**
+ * Class for interacting with the Filesystem, particularly, interacting with
+ * FRC-related paths on the system, such as the launch and deploy directories.
+ *
+ * <p>This class is primarily used for obtaining resources in src/main/deploy, and
+ * the RoboRIO path /home/lvuser in a simulation-compatible way.</p>
+ */
+public final class Filesystem {
+  private Filesystem() { }
+
+  /**
+   * Obtain the current working path that the program was launched with.
+   * This is analogous to the `pwd` command on unix.
+   *
+   * @return The current working directory (launch directory)
+   */
+  public static File getLaunchDirectory() {
+    return new File(".").getAbsoluteFile();
```

Comment:
@SamCarlberg Do you think it is better here to store a single instance as a class member and then return that instance each time?

Reply:
Yeah, there's no point in creating two new `File` objects every time
Also, I think the implementation here is wrong. The "." should either be an empty string or the user.dir system property.

```java
private static final File workingDir = new File(System.getProperty("user.dir"));
// Equivalent to this
private static final File workingDir = new File("");
```

I'd prefer using the user.dir property to be clear in what the file is, rather than relying on the default implementation of how the `File` class resolves relative paths

Reply:
Note that an empty string is not correct and will break when joining paths.